### PR TITLE
[spaceship] Spaceship::ConnectAPI - added betaGroup delete and get builds for the group functions

### PR DESCRIPTION
**Pull request by krish722**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Adding new features.

1. Delete TestFlight beta group.
2. Get builds for a TestFlight beta group.

### Description
1. Added a function delete! in BetaGroup class. We can call delete! will delete the group. It leaves the build as it is.
2. Added a function builds in BetaGroup class. We can call builds to get builds which are associated to the group.

### Testing Steps
```
Spaceship::ConnectAPI.login
platform = Spaceship::ConnectAPI::Platform.map('ios')
app = Spaceship::ConnectAPI::App.find('anyid')
group = app.get_beta_groups(filter: {name: "Gp1"}).first
group.fetch_builds.each do | build |
      puts  build.version
end
group.delete!
```

### CODE REVIEW

File: spaceship/lib/spaceship/connect_api/models/beta_group.rb [hunk 1]
```diff
@@ -36,6 +36,15 @@ def self.type
       def post_bulk_beta_tester_assignments(beta_testers: nil)
         return Spaceship::ConnectAPI.post_bulk_beta_tester_assignments(beta_group_id: id, beta_testers: beta_testers)
       end
+
+      def delete!
+        return Spaceship::ConnectAPI.delete_beta_group(group_id: id)
+      end
+
+      def builds
```
Comment (joshdholtz) [inline spaceship/lib/spaceship/connect_api/models/beta_group.rb:44]: Are you cool if we rename this to `fetch_builds`? I like to the method name as an action if its doing an API request 😇 

```suggestion
      def fetch_builds
```

**Review by joshdholtz — CHANGES_REQUESTED** [review]
This looks great! Just one small thought/request 😊

**Comment by krish722** [comment]
Please merge it. Also want to know when is the next release planned?

**Comment by joshdholtz** [comment]
@krish722 Next release should be later today or tomorrow morning 😊

**Review by joshdholtz — APPROVED** [review]
Thanks for making that change and for adding tests ❤️  Really appreciate the contribution!

**Pull request merged**